server/lib/s3backup/__init__.py

```diff
@@ -0,0 +1,176 @@
+                f.write(Body)
+
+
+class S3Config(object):
+    def __init__(self, config):
+        try:
+            debug_unittest = config.get('pbench-server-backup', 'debug_unittest')
+        except Exception as e:
+            debug_unittest = False
+        else:
+            debug_unittest = bool(debug_unittest)
+
+        self.endpoint_url = config.get('pbench-server-backup', 'endpoint_url')
+        self.bucket_name = config.get('pbench-server-backup', 'bucket_name')
+        self.access_key_id = config.get('pbench-server-backup', 'access_key_id')
+        self.secret_access_key = config.get('pbench-server-backup', 'secret_access_key')
+
+        if debug_unittest:
+            self.connector = MockS3Obj(self.endpoint_url, self.bucket_name)
+        else:
+            botoconnector = boto3.client('s3',
+                                   aws_access_key_id='{}'.format(self.access_key_id),
+                                   aws_secret_access_key='{}'.format(self.secret_access_key),
+                                   endpoint_url='{}'.format(self.endpoint_url))
+            self.connector = S3Obj(botoconnector)
+
+    def ts(self, *args):
+        return self.connector.ts(*args)
+
+    def upload_to_bucket(self, tarlist, report, _logger):
+        buck_check = self.connector.list_buckets()
+        if self.bucket_name in buck_check:
+            list_bucket_objects = self.connector.list_objects_v2(Bucket='{}'.format(self.bucket_name))
+            try:
+                s3_contents = [s3_content_obj['Key']
+                               for s3_content_obj in list_bucket_objects['Contents']]
+            except Exception:
+                s3_contents = []
+            tarlist.sort()
+            for tar in tarlist:
+                filename = os.path.basename(tar)
+                if filename in s3_contents:
+                    _logger.info(
+                        "Result: {} already exists\n".format(filename))
+                else:
+                    md5 = ("{}.md5".format(tar))
+                    with open(md5) as f:
+                        md5_hex_value = f.readline().split(" ")[0]
+                    md5_base64_value = (base64.b64encode(
+                        bytes.fromhex(md5_hex_value))).decode()
+                    try:
+                        data = open(tar, 'rb').read()
+                        self.connector.put_object(
+                            Bucket=self.bucket_name, Key=filename,
+                            Body=data, ContentMD5=md5_base64_value)
+                        _logger.info("Uploaded: {}".format(filename))
+                    except Exception as e:
+                        _logger.info(
+                            "Bad md5 for: {}, {}\n".format(filename, e))
+        else:
+            _logger.info(
+                "Bucket: {} is not present in s3\n".format(self.bucket_name))
+
+    def compare_backup_s3(self, s3_content_list, backup_content_list, report, _logger):
```

Comment:
This is really not part of the S3Config class: e.g. it does not use `self` at all. You should delete the `self` argument and pull it out of the class as a stand-alone utility routine. Actually, it's not used by `pbench-backup-tarballs` at all, so you could move it to `pbench-verify-backup-tarballs` which is its only user.